### ISSUE TITLE
Cursorline smears when scrolling with C-E/C-Y

### ISSUE THREAD

**Issue by rdavison** [neovide/neovide#2818]
**Describe the bug**
When cursorline is enabled and you place the cursor on the top or bottom edge of a window and then scroll with C-E or C-Y, then the cursorline will smear. See attached video.

**To Reproduce**
1. Enable `vim.opt.cursorline = true`
2. Place cursor at top of window
3. Press and hold C-E, you will see a smear

**Expected behavior**
The cursor line should only show on the line with the cursor, not on any other line.

**Screenshots**

https://github.com/user-attachments/assets/889a934b-2e95-42d0-865b-50073e581464

**Desktop (please complete the following information):**
 - OS: [e.g. macOS 14.7]
 - Neovide Version [e.g. 0.13.3]
 - Neovim Version [e.g. 0.10.0]

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
[neovide_rCURRENT.log](https://github.com/user-attachments/files/17180170/neovide_rCURRENT.log)

**Additional context**

**Comment by fredizzimo** [comment]
Yes, I noticed this when I was looking at the other [cursorline bug](https://github.com/neovide/neovide/issues/2794). 

This can't be fixed in Neovide alone, so I opened a feature request here:
* https://github.com/neovim/neovim/issues/30584
